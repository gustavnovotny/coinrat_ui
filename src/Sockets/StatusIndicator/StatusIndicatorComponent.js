// @flow
import React from 'react'
import {Intent, Icon} from "@blueprintjs/core"
import {Tooltip2} from "@blueprintjs/labs"

type Props = {
  isOnline: boolean,
}

const StatusIndicatorComponent = ({isOnline}: Props) => {
  const intent = isOnline ? Intent.SUCCESS : Intent.DANGER
  const tooltipContent = isOnline
    ? 'Connection with backend is just OK.'
    : 'Connection with backend cannot be established.'

  return (
    <Tooltip2 content={tooltipContent}>
      <Icon intent={intent} iconName="pt-icon-dot" iconSize="inherit"/>
    </Tooltip2>
  )
}

export default StatusIndicatorComponent
