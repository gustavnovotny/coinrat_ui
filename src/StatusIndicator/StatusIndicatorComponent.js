import React, {Component} from 'react'
import {Intent, Icon, Position, Tooltip} from "@blueprintjs/core"
import {observer} from "mobx-react"

const StatusIndicatorComponent = observer(class StatusIndicatorComponent extends Component {
  render() {
    const isOnline = this.props.store.isOnline

    const intent = isOnline ? Intent.SUCCESS : Intent.DANGER
    const tooltipContent = isOnline
      ? 'Connection with backend is just OK.'
      : 'Connection with backend cannot be established.'

    return (

      <Tooltip content={tooltipContent} position={Position.BOTTOM_LEFT}>
        <Icon intent={intent} iconName="pt-icon-dot" iconSize="inherit"/>
      </Tooltip>
    )
  }
})

export default StatusIndicatorComponent
