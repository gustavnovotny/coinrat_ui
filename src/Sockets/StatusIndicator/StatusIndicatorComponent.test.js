// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import StatusIndicatorComponent from "./StatusIndicatorComponent"

it('renders correctly, ONLINE', () => {
  const tree = renderer.create(<StatusIndicatorComponent isOnline={true}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders correctly, OFFLINE', () => {
  const tree = renderer.create(<StatusIndicatorComponent isOnline={false}/>).toJSON()
  expect(tree).toMatchSnapshot()
})
