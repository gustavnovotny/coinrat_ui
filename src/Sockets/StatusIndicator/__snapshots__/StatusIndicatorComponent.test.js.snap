// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly, OFFLINE 1`] = `
<span>
  <div
    className="pt-popover-target"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="pt-icon pt-icon-dot pt-intent-danger"
      disabled={undefined}
      tabIndex={0}
    />
  </div>
</span>
`;

exports[`renders correctly, ONLINE 1`] = `
<span>
  <div
    className="pt-popover-target"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <span
      className="pt-icon pt-icon-dot pt-intent-success"
      disabled={undefined}
      tabIndex={0}
    />
  </div>
</span>
`;
